File: service/service/report/service_details_report/service_details_report.js
// Copyright (c) 2022, thirvusoft@gmail.com and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Service Details Report"] = {
	"filters": [
		{
			"fieldname":"from_date",
			"label": __("From Date"),
			"fieldtype": "Date",
			"default": frappe.datetime.get_today(),
			"reqd":1,
			"width": "80"
		},
		{
			"fieldname":"to_date",
			"label": __("To Date"),
			"fieldtype": "Date",
			"reqd":1,
			"default": frappe.datetime.get_today(),
			"width": "80"
		},
		{
			"fieldname":"status",
			"label": __("Status"),
			"fieldtype": "Select",
			"options": "Open\nOn Process\nCompleted\nReady For Delivery\nOverdue\nGoods Delivered\nCancelled",
			"width": "100"
		},
		

	]
};

File: service/service/report/service_details_report/service_details_report.py
# Copyright (c) 2022, thirvusoft@gmail.com and contributors
# For license information, please see license.txt

import frappe




def execute(filters=None):
	from_date = filters.get("from_date")
	to_date = filters.get("to_date")
	status = filters.get("status")
	
	filters = []
	filters.append(["status", "=",status])
	
	if to_date:
	
		filters.append(["posting_date", "<=", to_date])
	if from_date:
		filters.append(["posting_date", ">=", from_date])

	
	
	result_data=frappe.get_list("Service",fields=["customer","posting_date","expected_delivery_date","status","service_item","item_count","received_employee","total_amount","advance_amount"],	filters=filters)
						
			
	columns = get_columns()
	return columns,result_data
 
def get_columns():
	columns = [
		("Customer")  + ":Data:100",
		("Posting Date") + ":Data:100",
		("Expected Delivery Date") + ":Link/Designation:200",
		("Status") + ":Data:100",
		("Service Item") + ":Data:100",
		("Item Count") + ":Data:100",
		("Received Employee") + ":Data:100",
		("Total Amount") + ":Currency:100",
		("Paid Amount") + ":Currency:100",
		
		]
	
	return columns
	
